src/components/legend/get_legend_data.js

```diff
@@ -71,21 +73,34 @@ module.exports = function getLegendData(calcdata, opts) {
     var ltraces;
     var legendData;
 
+    // sort considering trace.legendrank and legend.traceorder
+    var dir = helpers.isReversed(opts) ? -1 : 1;
+    var orderFn = function(a, b) {
+        var A = a[0].trace;
+        var B = b[0].trace;
+        var delta = A.legendrank - B.legendrank;
+        if(!delta) delta = A.index - B.index;
+        if(!delta) delta = a[0]._initID - b[0]._initID;
```

Comment:
I'm assuming `_initID` is important for pie traces? Anything else? OK, so the code below implies (and I just tested to confirm) that `legend.traceorder` flips the order of pie entries in the legend, not just the order individual traces get added to the legend.

Based on that, I'd say the correct behavior of traceorder once we add `legendrank` is to simply flip the whole thing. That's not what I thought we concluded on the call yesterday but it's what you've implemented and with the pie complication I think it's the simplest thing to explain :)

Reply:
I'm fine with this.

Reply:
> I'm assuming `_initID` is important for pie traces? Anything else? OK, so the code below implies (and I just tested to confirm) that `legend.traceorder` flips the order of pie entries in the legend, not just the order individual traces get added to the legend.

I noticed some older version of Chrome (including our image test) sort the items differently. 
So `_initID` was added in https://github.com/plotly/plotly.js/pull/5591/commits/a60a382ee0c72d5c50dc5cb36d7c4e3766c4ff42 to address that issue.

As you mentioned `pie-like` traces appear to behave differently.
In https://github.com/plotly/plotly.js/pull/5591/commits/598d60c9153baac1f03a2691554752a778d1d4a1
I was expecting that by using a higher value of `legendrank`on the first `pie` trace the slice legends come next.
However it is the opposite and it is rather confusing.